x-pack/plugins/fleet/public/applications/fleet/components/fleet_server_instructions/index.tsx

```diff
@@ -0,0 +1,154 @@
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import styled from 'styled-components';
+
+import { useStartServices } from '../../hooks';
+
+import { QuickStartTab } from './quick_start_tab';
+import { AdvancedTab } from './advanced_tab';
+
+const ContentWrapper = styled(EuiFlexGroup)`
+  height: 100%;
+  margin: 0 auto;
+`;
+
+interface Props {
+  onClose: () => void;
+}
+
+const useFleetServerTabs = () => {
+  const [currentTab, setCurrentTab] = useState('quickStart');
+
+  const quickStartTab = {
+    id: 'quickStart',
+    name: 'Quick Start',
+    content: <QuickStartTab />,
+  };
+
+  const advancedTab = {
+    id: 'advanced',
+    name: 'Advanced',
+    content: <AdvancedTab />,
+  };
+
+  const currentTabContent =
+    currentTab === 'quickStart' ? quickStartTab.content : advancedTab.content;
+
+  return { tabs: [quickStartTab, advancedTab], currentTab, setCurrentTab, currentTabContent };
+};
+
+const Header: React.FunctionComponent<{
+  isFlyout?: boolean;
+  currentTab: string;
+  tabs: Array<{ id: string; name: string; content: React.ReactNode }>;
+  onTabClick: (id: string) => void;
+}> = ({ isFlyout = false, currentTab: currentTabId, tabs, onTabClick }) => {
+  const { docLinks } = useStartServices();
+
+  return (
+    <>
+      <EuiTitle size="m">
+        <h2>
+          <FormattedMessage
+            id="xpack.fleet.fleetServerFlyout.title"
+            defaultMessage="Add a Fleet Server"
+          />
+        </h2>
+      </EuiTitle>
+
+      <EuiSpacer size="m" />
+
+      <EuiText>
+        <FormattedMessage
+          id="xpack.fleet.fleetServerFlyout.instructions"
+          defaultMessage="A Fleet Server is required before you can enroll agents with Fleet. Follow the instructions below to set up a Fleet Server. For more information, see the {userGuideLink}"
+          values={{
+            userGuideLink: (
+              <EuiLink
+                href={docLinks.links.fleet.fleetServerAddFleetServer}
+                external
+                target="_blank"
+              >
+                <FormattedMessage
+                  id="xpack.fleet.fleetServerSetup.setupGuideLink"
+                  defaultMessage="Fleet and Elastic Agent Guide"
+                />
+              </EuiLink>
+            ),
+          }}
+        />
+      </EuiText>
+
+      <EuiSpacer size="m" />
+
+      <EuiTabs style={{ marginBottom: isFlyout ? '-25px' : '' }}>
+        {tabs.map((tab) => (
+          <EuiTab
+            key={`fleetServerFlyoutTab-${tab.id}`}
+            data-test-subj={`fleetServerFlyoutTab-${tab.id}`}
+            id={tab.id}
+            isSelected={tab.id === currentTabId}
+            onClick={() => onTabClick(tab.id)}
+          >
+            {tab.name}
+          </EuiTab>
+        ))}
+      </EuiTabs>
+    </>
+  );
+};
+
+// Renders instructions inside of a flyout
+export const FleetServerFlyout: React.FunctionComponent<Props> = ({ onClose }) => {
```

Comment:
This file exports both the flyout "version" of this component as well as the non-flyout version. It seemed a bit extreme to separate these into two files, and then share the tab implementation between those files, but it's something we might want to consider.